cmd/function-sidecar.go

```diff
@@ -27,59 +26,83 @@ import (
 	"github.com/bsm/sarama-cluster"
 	"gopkg.in/Shopify/sarama.v1"
 
+	"flag"
+	dispatch "github.com/projectriff/function-sidecar/pkg/dispatcher"
+	"github.com/projectriff/function-sidecar/pkg/dispatcher/grpc"
 	"github.com/projectriff/function-sidecar/pkg/dispatcher/http"
 	"github.com/projectriff/function-sidecar/pkg/dispatcher/stdio"
-	"github.com/projectriff/function-sidecar/pkg/dispatcher"
 	"github.com/projectriff/function-sidecar/pkg/message"
-	"github.com/projectriff/function-sidecar/pkg/dispatcher/grpc"
+	"strings"
 )
 
+type stringSlice []string
+
+func (sl *stringSlice) String() string {
+	return fmt.Sprint(*sl)
+}
+
+func (sl *stringSlice) Set(value string) error {
+	*sl = stringSlice(strings.Split(value, ","))
+	return nil
+}
+
+var brokers stringSlice = []string{"localhost:9092"}
+var inputs, outputs stringSlice
+var group, protocol string
+
+func init() {
+	flag.Var(&brokers, "brokers", "location of the Kafka server(s) to connect to")
+	flag.Var(&inputs, "inputs", "kafka topic(s) to listen to, as input for the function")
+	flag.Var(&outputs, "outputs", "kafka topic(s) to write to with results from the function")
+	flag.StringVar(&group, "group", "", "kafka consumer group to act as")
+	flag.StringVar(&protocol, "protocol", "", "dispatcher protocol to use. One of [http, grpc, stdio]")
+}
+
 func main() {
-	var saj map[string]interface{}
-	err := json.Unmarshal([]byte(os.Getenv("SPRING_APPLICATION_JSON")), &saj)
-	if err != nil {
-		panic(err)
-	}
 
-	consumerConfig := makeConsumerConfig()
-	consumerConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	flag.Parse()
 
-	brokers := []string{saj["spring.cloud.stream.kafka.binder.brokers"].(string)}
-	input := saj["spring.cloud.stream.bindings.input.destination"].(string)
-	output := saj["spring.cloud.stream.bindings.output.destination"]
-	group := saj["spring.cloud.stream.bindings.input.group"].(string)
-	protocol := saj["spring.profiles.active"].(string)
+	input := inputs[0]
```

Comment:
since we are only using the first input and the first output, why are we using a slice?

Reply:
We could use a single valued var, but if we aim to support multiple inputs/outputs (and I think we said we'd do) I wanted the exposed name of the arg to reflect that _now_, as it's part of the "API" of the sidecar.
You might argue that this is not user exposed though :/

Reply:
can we log a warning if `len > 1` stating that multiple destinations is not yet supported?

Reply:
We sure can (even though I'm not sure what "warning" really means with the basic go log library). As a matter of fact, I think it's even ok to panic completely

Reply:
sure that would work, and let's create an issue for multiple destinations now if it doesn't exist yet (could even refer to that issue in the log message when we panic)

Reply:
Created https://github.com/projectriff/riff/issues/184 as this obviously spans more than just the sidecar. will update the code with a ref to that URL